txtorcon/circuit.py

```diff
@@ -15,14 +15,34 @@
 from twisted.internet.interfaces import IStreamClientEndpoint
 from zope.interface import implementer
 
-from .interface import IRouterContainer, IStreamAttacher
+from .interface import IRouterContainer, IStreamAttacher, ICircuitListener
 from txtorcon.util import find_keywords, maybe_ip_addr, SingleObserver
 
+from txtorcon.log import txtorlog
+
 
 # look like "2014-01-25T02:12:14.593772"
 TIME_FORMAT = '%Y-%m-%dT%H:%M:%S'
 
 
+def extract_reason(kw):
+    """
```

Comment:
I'd rather keep the name with a leading underscore (so it's "private" and we don't have to care about the API's backwards-compat).